iree/turbine/kernel/wave/compile.py

```diff
@@ -32,16 +34,26 @@ def invoke(self, *args, **kwargs):
         Returns the assembly code of the compiled kernel.
         """
 
+        # Segregate args into kernel tensor and scalars.
+        tensor_args, scalar_args = [], []
+
+        for arg in args:
+            (tensor_args if isinstance(arg, torch.Tensor) else scalar_args).append(arg)
+
+        tensor_args = tuple(tensor_args)
+        scalar_args = tuple(scalar_args)
+
+        kernel_inputs, kernel_outputs = [], []
         # Partition arguments into kernel inputs and outputs.
-        kernel_inputs = []
-        kernel_outputs = []
-        for arg, usage in zip(args, self.options.kernel_usages):
+        for arg, usage in zip(tensor_args, self.options.kernel_usages):
             if usage == kernel_codegen.KernelBufferUsage.INPUT:
```

Comment:
Any thoughts on combinining these two loops together? and just have a special case here if it's a scalar we add to scalar_args instead of kernel_inputs. Then at the end we just set kerneL_inputs.extend(scalar_args)

Reply:
loop is kind of expensive in python, so ideally we don't bring extra there

Reply:
The logic was split into two loops because we are using `zip` and `scalar args` do not have `kernel_usages`. 
So, it will lead to incorrect result.

Reply:
Addressed